Modules/_localemodule.c

```diff
@@ -692,7 +692,17 @@ _locale_nl_langinfo_impl(PyObject *module, int item)
             result = result != NULL ? result : "";
             char *oldloc = NULL;
             if (langinfo_constants[i].category != LC_CTYPE
-                && !is_all_ascii(result)
+                // gh-133740: Always change the locale for ALT_DIGITS and ERA
+                && (
+#ifdef __GLIBC__
```

Comment:
Or maybe move the comment even more down. In any way LGTM.
```suggestion
                && (
#ifdef __GLIBC__
                    // gh-133740: Always change the locale for ALT_DIGITS and ERA
```